detect/master_changer_test.go

```diff
@@ -0,0 +1,133 @@
+package detect
+
+import (
+	"reflect"
+	"testing"
+
+	mesos "github.com/mesos/mesos-go/mesosproto"
+	"github.com/mesosphere/mesos-dns/logging"
+)
+
+func TestMasters_UpdatedMasters(t *testing.T) {
+	// create a new master changer with an unknown leader and no masters
```

Comment:
Do you mean from `master_changer.go` maybe to `masters_detector.go`? If so, sure, that's a very good suggestion :-)